app/Jobs/ApplicationDeploymentJob.php

```
@@ -2463,35 +2741,189 @@ private function add_build_env_variables_to_dockerfile()
+                }
+            }
+        }
+
+        // Update the compose file
+        $composeFile['services'] = $services;
+        $composeFile['secrets'] = $secrets;
+
+        $this->application_deployment_queue->addLogEntry('Added build secrets configuration to docker-compose file.');
+
+        return $composeFile;
+    }
```

PHPMD (2.15.0)

The method add_build_secrets_to_compose() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)